src/plugins/unified_histogram/public/container/services/state_service.ts

```diff
@@ -0,0 +1,226 @@
+  filters: Filter[];
+  /**
+   * The current Lens request adapter
+   */
+  lensRequestAdapter: RequestAdapter | undefined;
+  /**
+   * The current query
+   */
+  query: Query | AggregateQuery;
+  /**
+   * The current request adapter used for non-Lens requests
+   */
+  requestAdapter: RequestAdapter | undefined;
+  /**
+   * The current search session ID
+   */
+  searchSessionId: string | undefined;
+  /**
+   * The current time interval of the chart
+   */
+  timeInterval: string;
+  /**
+   * The current time range
+   */
+  timeRange: TimeRange;
+  /**
+   * The current top panel height
+   */
+  topPanelHeight: number | undefined;
+  /**
+   * The current fetch status of the hits count request
+   */
+  totalHitsStatus: UnifiedHistogramFetchStatus;
+  /**
+   * The current result of the hits count request
+   */
+  totalHitsResult: number | Error | undefined;
+}
+
+/**
+ * The options used to initialize the comntainer state
+ */
+export interface UnifiedHistogramStateOptions {
+  /**
+   * The services required by the Unified Histogram components
+   */
+  services: UnifiedHistogramServices;
+  /**
+   * The prefix for the keys used in local storage -- leave undefined to avoid using local storage
+   */
+  localStorageKeyPrefix?: string;
+  /**
+   * The initial state of the container
+   */
+  initialState: Partial<UnifiedHistogramState> & Pick<UnifiedHistogramState, 'dataView'>;
+}
+
+export type UnifiedHistogramStateService = Omit<ReturnType<typeof createStateService>, 'state$'> & {
+  state$: Observable<UnifiedHistogramState>;
+};
+
+export const createStateService = (options: UnifiedHistogramStateOptions) => {
+  const { services, localStorageKeyPrefix, initialState } = options;
+
+  let initialChartHidden = false;
+  let initialTopPanelHeight: number | undefined;
+  let initialBreakdownField: string | undefined;
+
+  if (localStorageKeyPrefix) {
+    initialChartHidden = getChartHidden(services.storage, localStorageKeyPrefix) ?? false;
+    initialTopPanelHeight = getTopPanelHeight(services.storage, localStorageKeyPrefix);
+    initialBreakdownField = getBreakdownField(services.storage, localStorageKeyPrefix);
+  }
+
+  const state$ = new BehaviorSubject({
+    breakdownField: initialBreakdownField,
+    chartHidden: initialChartHidden,
+    filters: [],
+    lensRequestAdapter: undefined,
+    query: services.data.query.queryString.getDefaultQuery(),
+    requestAdapter: undefined,
+    searchSessionId: undefined,
+    timeInterval: 'auto',
+    timeRange: services.data.query.timefilter.timefilter.getTimeDefaults(),
+    topPanelHeight: initialTopPanelHeight,
+    totalHitsResult: undefined,
+    totalHitsStatus: UnifiedHistogramFetchStatus.uninitialized,
+    ...initialState,
+  });
+
+  const updateState = (stateUpdate: Partial<UnifiedHistogramState>) => {
+    state$.next({
+      ...state$.getValue(),
+      ...stateUpdate,
+    });
+  };
+
+  return {
+    /**
+     * The current state of the container
+     */
+    state$,
```

Comment:
As best I can tell this makes all `state` externally mutable. It would be better to make this `Omit<BehaviorSubject, 'next'>`

Reply:
Yup it would have, good catch. I updated the interface for this and am exposing it as `Observable<UnifiedHistogramState>` now.